framework_tests/test_couchbase_server_install.py

```diff
@@ -0,0 +1,35 @@
+from libraries.provision.install_couchbase_server import CouchbaseServerConfig
+
+
+def test_version_only():
+    server_config = CouchbaseServerConfig(
+        version="4.1.1"
+    )
+    assert server_config.version == "4.1.1"
+    assert server_config.build is None
+
+
+def test_version_build():
+    server_config = CouchbaseServerConfig(
+        version="4.1.1-5914"
+    )
+    assert server_config.version == "4.1.1"
+    assert server_config.build == "5914"
+
+
+def test_enterprise():
+    server_config = CouchbaseServerConfig(
+        version="4.1.1-5914"
+    )
+    base_url, package_name = server_config.get_baseurl_package()
+    assert base_url == "http://cbnas01.sc.couchbase.com/builds/latestbuilds/couchbase-server/sherlock/5914"
```

Comment:
We depend on a hardcoded url value in the provisioning code so if that changes we could just change the to reflect the new download url? I really have no strong feelings either way. I just feel like this is more explicit